### PR TITLE
MAINT: refactor DBSCAN onedal support funcs

### PR DESCRIPTION
**Pull request by samir-nasibli**
# Description
refactor DBSCAN onedal support funcs:
* added common `_onedal_supported` function to avoid code duplication 
* Addressed  https://github.com/intel/scikit-learn-intelex/pull/1400#discussion_r1307959551
* added patching status

### CODE REVIEW

File: sklearnex/cluster/dbscan.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def _onedal_fit(self, X, y, sample_weight=None, queue=None):
         self._onedal_estimator.fit(X, y=y, sample_weight=sample_weight, queue=queue)
         self._save_attributes()
 
-    def _onedal_cpu_supported(self, method_name, *data):
+    def _onedal_supported(self, method_name, *data):
```
Comment (Alexsandruss) [inline sklearnex/cluster/dbscan.py:101]: Patching verbosity should be added.

**Review by Alexsandruss — CHANGES_REQUESTED** [review]
Change `BaseDBSCANspmd` and `DBSCAN_Batch` names to ones with correct python letter cases in onedal/spmd/cluster/dbscan.py and sklearnex/spmd/cluster/dbscan.py

File: sklearnex/cluster/dbscan.py (hunk 1 above)
Comment (samir-nasibli) [reply sklearnex/cluster/dbscan.py:101]: Please read the description to the draft pr properly.

**Comment by samir-nasibli** [comment]
> Change `BaseDBSCANspmd` and `DBSCAN_Batch` names to ones with correct python letter cases in onedal/spmd/cluster/dbscan.py and sklearnex/spmd/cluster/dbscan.py

It is consistent with other spmd algos. There is no python issues. If your have any suggestion please share.

**Comment by samir-nasibli** [comment]
GPU CI Job: http://intel-ci.intel.com/ee468036-46c6-f142-9fc7-a4bf010d0e2e. Specifically for sklearnex/daal4py tests/examples on GPU.

**Review by ethanglaser — APPROVED** [review]

**Pull request merged**